nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/main/java/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQL.java

```diff
@@ -974,139 +938,92 @@ public void outputEvents(ProcessSession session, ComponentLog log) throws IOExce
+                        binlogResourceInfo.setInTransaction(false);
+                        updateState(session, dataCaptureState);
                         // If there is no FlowFile open, commit the session
                         if (eventWriterConfiguration.getCurrentFlowFile() == null) {
                             // Commit the NiFi session
                             session.commitAsync();
                         }
-                        currentTable = null;
+                        binlogResourceInfo.setCurrentTable(null);
+                        binlogResourceInfo.setCurrentDatabase(null);
                     } else {
                         // Check for DDL events (alter table, e.g.). Normalize the query to do string matching on the type of change
                         String normalizedQuery = normalizeQuery(sql);
 
-                        if (normalizedQuery.startsWith("alter table")
-                                || normalizedQuery.startsWith("alter ignore table")
-                                || normalizedQuery.startsWith("create table")
-                                || normalizedQuery.startsWith("truncate table")
-                                || normalizedQuery.startsWith("rename table")
-                                || normalizedQuery.startsWith("drop table")
-                                || normalizedQuery.startsWith("drop database")) {
-
-                            if (includeDDLEvents && (databaseNamePattern == null || databaseNamePattern.matcher(currentDatabase).matches())) {
-                                // If we don't have table information, we can still use the database name
-                                TableInfo ddlTableInfo = (currentTable != null) ? currentTable : new TableInfo(currentDatabase, null, null, null);
-                                DDLEventInfo ddlEvent = useGtid
-                                        ? new DDLEventInfo(ddlTableInfo, timestamp, currentGtidSet, sql)
-                                        : new DDLEventInfo(ddlTableInfo, timestamp, currentBinlogFile, currentBinlogPosition, sql);
-                                currentEventInfo = ddlEvent;
-                                currentEventWriter = ddlEventWriter;
-                                currentSequenceId.set(ddlEventWriter.writeEvent(currentSession, transitUri, ddlEvent, currentSequenceId.get(), REL_SUCCESS, eventWriterConfiguration));
-                            }
-                            // Remove all the keys from the cache that this processor added
-                            if (cacheClient != null) {
-                                cacheClient.removeByPattern(this.getIdentifier() + ".*");
+                        if (isQueryDDL(normalizedQuery)) {
+                            if (!(databaseNamePattern != null && !databaseNamePattern.matcher(binlogResourceInfo.getCurrentDatabase()).matches())) {
```

Comment:
This can be simplified.
```suggestion
                            if (databaseNamePattern == null || databaseNamePattern.matcher(binlogResourceInfo.getCurrentDatabase()).matches())) {
```